### ISSUE TITLE
uring fails to compile on RHEL 7.x

### ISSUE THREAD

**Issue by mikeBashStuff** [spdk/spdk#1564]
```
SPDK:  v20.07-rc1-190-g5adf099
OS: RHEL 7.8 (Kernel: 3.10.0, gcc: 4.8.5)
```

`pkgdep.sh` fails to install uring due to the following:
```
include/liburing/barrier.h:56:23: fatal error: stdatomic.h: No such file or directory
 #include <stdatomic.h>
                       ^
compilation terminated.
make[1]: *** [setup.ol] Error 1
make[1]: Leaving directory `/usr/local/src/liburing/src'
make: *** [install] Error 2
```
However, even with the missing header file in place, compilation still fails:

```
include/liburing/barrier.h:66:25: error: ‘_Atomic’ undeclared (first use in this function)
  atomic_store_explicit((_Atomic typeof(*(p)) *)(p), (v), \
                         ^
include/liburing.h:150:3: note: in expansion of macro ‘io_uring_smp_store_release’
   io_uring_smp_store_release(cq->khead, *cq->khead + nr);
   ^
include/liburing/barrier.h:66:25: note: each undeclared identifier is reported only once for each function it appears in
  atomic_store_explicit((_Atomic typeof(*(p)) *)(p), (v), \
                         ^
include/liburing.h:150:3: note: in expansion of macro ‘io_uring_smp_store_release’
   io_uring_smp_store_release(cq->khead, *cq->khead + nr);
   ^
include/liburing/barrier.h:66:33: error: expected ‘)’ before ‘typeof’
  atomic_store_explicit((_Atomic typeof(*(p)) *)(p), (v), \
                                 ^
include/liburing.h:150:3: note: in expansion of macro ‘io_uring_smp_store_release’
   io_uring_smp_store_release(cq->khead, *cq->khead + nr);
   ^
include/liburing/barrier.h:66:33: error: incompatible type for argument 1 of ‘__atomic_store_n’
  atomic_store_explicit((_Atomic typeof(*(p)) *)(p), (v), \
                                 ^
include/liburing.h:150:3: note: in expansion of macro ‘io_uring_smp_store_release’
   io_uring_smp_store_release(cq->khead, *cq->khead + nr);
   ^
include/liburing.h: In function ‘io_uring_cq_ready’:
include/liburing/barrier.h:69:24: error: ‘_Atomic’ undeclared (first use in this function)
  atomic_load_explicit((_Atomic typeof(*(p)) *)(p), \
                        ^
include/liburing.h:451:9: note: in expansion of macro ‘io_uring_smp_load_acquire’
  return io_uring_smp_load_acquire(ring->cq.ktail) - *ring->cq.khead;
         ^
include/liburing/barrier.h:69:32: error: expected ‘)’ before ‘typeof’
  atomic_load_explicit((_Atomic typeof(*(p)) *)(p), \
                                ^
include/liburing.h:451:9: note: in expansion of macro ‘io_uring_smp_load_acquire’
  return io_uring_smp_load_acquire(ring->cq.ktail) - *ring->cq.khead;
         ^
include/liburing/barrier.h:69:32: error: incompatible type for argument 1 of ‘__atomic_load_n’
  atomic_load_explicit((_Atomic typeof(*(p)) *)(p), \
                                ^
include/liburing.h:451:9: note: in expansion of macro ‘io_uring_smp_load_acquire’
  return io_uring_smp_load_acquire(ring->cq.ktail) - *ring->cq.khead;
         ^
include/liburing.h: In function ‘io_uring_cq_eventfd_toggle’:
include/liburing/barrier.h:59:25: error: ‘_Atomic’ undeclared (first use in this function)
  atomic_store_explicit((_Atomic typeof(var) *)&(var), \
                         ^
include/liburing.h:480:2: note: in expansion of macro ‘IO_URING_WRITE_ONCE’
  IO_URING_WRITE_ONCE(*ring->cq.kflags, flags);
  ^
include/liburing/barrier.h:59:33: error: expected ‘)’ before ‘typeof’
  atomic_store_explicit((_Atomic typeof(var) *)&(var), \
                                 ^
include/liburing.h:480:2: note: in expansion of macro ‘IO_URING_WRITE_ONCE’
  IO_URING_WRITE_ONCE(*ring->cq.kflags, flags);
  ^
include/liburing/barrier.h:59:33: error: incompatible type for argument 1 of ‘__atomic_store_n’
  atomic_store_explicit((_Atomic typeof(var) *)&(var), \
                                 ^
include/liburing.h:480:2: note: in expansion of macro ‘IO_URING_WRITE_ONCE’
  IO_URING_WRITE_ONCE(*ring->cq.kflags, flags);
  ^
make[1]: *** [setup.ol] Error 1
make[1]: Leaving directory `/usr/local/src/liburing/src'
make: *** [install] Error 2
```
gcc and the kernel in this distro are quite old though, so the question is if such a setup can be considered as supported.

**Comment by optimistyzy** [comment]
@mikeBashStuff I see that your kernel version is : 3.10.0, so clearly uring will not be supported in this kernel version. From my mind, we can close this issue.

**Comment by mikeBashStuff** [comment]
I think we should at least consider adding some warning into `pkgdep.sh` about the kernel version we try to build uring against so user is aware what's going on with regards to `--uring` arg. This can be handled as a separate task though, closing this issue. :)
